### ISSUE TITLE
Proposal: Revamp and Migrate rules_antlr to bazel-contrib

### ISSUE THREAD

**Issue by albertocavalcante** [marcohu/rules_antlr#25]
## TL;DR
The canonical reference to `rules_antlr` is lost. The proposal is to transfer it to [bazel-contrib](https://github.com/bazel-contrib). Help wanted.

## Context

I'm creating this issue as @marcohu hasn't been active on GitHub since 2022, and this repo hasn't been maintained for much longer (as evidenced by [issue #19](https://github.com/marcohu/rules_antlr/issues/19)).

The only other contributor besides @marcohu is @vmax, who I believe does not have write access and also has stale PRs ([see PR #18](https://github.com/marcohu/rules_antlr/pull/18)).

I've also tried to find ["Marco Hunsicker"](https://github.com/marcohu/rules_antlr/blob/master/src/main/java/org/antlr/bazel/AntlrRules.java#L37C13-L37C19) on LinkedIn without success. If anyone knows how to contact him, please let me know.

That said, there are some [forks](https://github.com/marcohu/rules_antlr/forks) available, but they do not inspire confidence in terms of maintainability and are not suitable for various use cases. These forks seem to be one-time efforts to add specific features or support for certain ANTLR versions and haven't been updated since.

## Bzlmod

In its current state, the project can't be published to the Bazel Central Registry (BCR), and projects relying on it won't be able to leverage Bzlmod to pull this dependency. There's an issue requesting its inclusion ([see issue #1575](https://github.com/bazelbuild/bazel-central-registry/issues/1575)), and another project is awaiting its inclusion ([see issue #137](https://github.com/woven-planet/go-zserio/issues/137)).

## Proposal

Following @EdSchouten's suggestion ([see issue comment](https://github.com/marcohu/rules_antlr/issues/19#issuecomment-2029997441)), I propose a "hostile takeover" to migrate these rules to @bazel-contrib.

The guidance is provided at [Hosting rules repositories in the bazel-contrib org](https://bazel-contrib.github.io/SIG-rules-authors/hosting-policy.html), which states:

> Criteria for accepting rule sets into this repo and avoid them from getting stale:
> 
> 1. Must use an [OSI-approved OSS license](https://opensource.org/licenses), preferably [Apache-2.0](https://www.apache.org/licenses/LICENSE-2.0).
> 1. Must have wide applicability in the community.
> 1. Must have a clear point of contact who answers questions from the SIG.
> 1. Must be "production quality":
>     - clear README or other documentation outlining the goal of these rules, how to use them etc.
>     - generated API documentation
>     - include examples of use
>     - tests that are running continuously
> 1. Must reply to issues/PRs in 2-3 weeks (exact service level agreement TBD)
> 1. Must have more than one person who is committed to review/approve PRs
>     - We recommend encoding this as a `CODEOWNERS` file.
> 1. Must publish semver releases.
>     - Optional: follow the same release pattern as the rules-template does.
> 1. Must work with LTS Bazel version
> 1. Must publish the rules to the [Bazel Central Registry](https://github.com/bazelbuild/bazel-central-registry), keep that CI green

## Call to Action

The project clearly does not comply with all these items. In the meantime, I've created a fork at [my repository](https://github.com/albertocavalcante/rules_antlr), and my intention is to uplift it, reconcile it with other forks (when possible), and hand it over.

However, I would love not to do this alone, so I'm asking for help or volunteers.

I also want to highlight items 5 and 6, which state that we will need people willing to review and approve PRs.

**Comment by spags-lacework** [comment]
I've had to fork and add support to 4.13.1, as some of the forks support and some don't. I'd like to try help where I can to support.

**Comment by lalten** [comment]
Any update on this? Looking through the committer emails there is `Marco Hunsicker <bazel_rules@hunsicker.de>`, have you tried to mail him?

**Comment by albertocavalcante** [comment]
> Any update on this? Looking through the committer emails there is `Marco Hunsicker <bazel_rules@hunsicker.de>`, have you tried to mail him?

```txt
From: alberto@cavalcante.uk
Date: Aug 28, 2024, 3:52 PM -0300
To: bazel_rules@hunsicker.de


Hi Marcos,

Hope you’re doing well.

Just wondering if I can reach out to you over this email. Thanks!


Alberto
```

@lalten just tried with no luck.

The update is that I'm actively working on this but on a best effort basis. PRs are welcome!

Also, @mortenmj raised this PR for BCR to include Bzlmod support: https://github.com/bazelbuild/bazel-central-registry/pull/2650

I intend to take a look and incorporate these changes this week. Right now I'm still trying to upgrade `rules_antlr` to build with Bazel 6+ without breaking the tests.

**Comment by albertocavalcante** [comment]
Hi @spags-lacework, I tried to look for your `rules_antlr` fork to open a PR against mine but couldn't find it https://github.com/spags-lacework?tab=repositories&q=&type=fork&language=&sort=. Maybe it's private? 

I was wondering if you could raise so I can incorporate your changes. Thanks!

**Comment by spags-lacework** [comment]
Sure, it's at https://github.com/lacework/rules_antlr/commit/4b0bd2ed141f961f3c0142bb700e361c4ae4e106.
There's a couple of todo's, the premise is refactoring the helper methods to be version independent and then providing the new list of versions.

**Comment by amahpour** [comment]
Also interested here as well. Can this be used with the latest version of Bazel?

**Comment by albertocavalcante** [comment]
@amahpour It can be used in the latest version, but still in `WORKSPACE` mode, via `--enable_workspace`. 

Bzlmod support is WIP. @peakschris has a PR at https://github.com/albertocavalcante/rules_antlr/pull/35 and also via these patches it seems to provide an extension function to use with `MODULE.bazel`: https://github.com/bazelbuild/bazel-central-registry/pull/2650/files#diff-691730f6c8594c59654e38e4b24079e9bd9d83f5e4bd45d6e1b05601899efa58

re: bzlmod enablement, better have the discussion at https://github.com/albertocavalcante/rules_antlr/issues/34
